In [5]:
import numpy as np 
import pandas as pd 

In [6]:
import pandas as pd
import numpy as np
import matplotlib 
import matplotlib.pyplot as plt
import seaborn as sns

# Data Preperation & Cleaning

In [14]:
df = pd.read_csv('./AB_NYC_2019.csv')

df

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

In [8]:
print(f"Rows: {df.shape[0]}, Columns: {df.shape[1]}")

Rows: 48895, Columns: 16


In [9]:
print(df.isnull().sum())

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64


name has null values  last review has and so reviies per month

In [15]:
print(f"Number of rows: {df.shape[0]}")
print(f"Number of columns: {df.shape[1]}")

Number of rows: 48895
Number of columns: 16


In [18]:
df.replace([np.inf, -np.inf], np.nan, inplace=True)

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [20]:
df.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

In [22]:
df.fillna({'reviews_per_month': 0}, inplace=True)
df.dropna(subset=['name', 'host_name'], inplace=True)

In [23]:
missing_values = df.isnull().sum()
print(missing_values[missing_values > 0])

last_review    10037
dtype: int64


In [24]:
df = df.dropna()

In [ ]:
print(df.isnull().sum())

In [ ]:
df.replace([np.inf, -np.inf], np.nan, inplace=True)
df.fillna({'reviews_per_month': 0}, inplace=True)
df.dropna(subset=['name', 'host_name'], inplace=True)

# Exploratory analysis & visualization

In [ ]:
basic_stats = df.describe()
print(basic_stats)

range_stats = df.max(numeric_only=True) - df.min(numeric_only=True)
print("Range of numeric columns:")
print(range_stats)

In [ ]:
df.describe()

In [ ]:
plt.figure(figsize=(10, 6))
sns.histplot(df['price'], bins=50, kde=True)
plt.title('Distribution of Price')
plt.xlabel('Price')
plt.ylabel('Frequency')
plt.show()

In [ ]:
plt.figure(figsize=(10, 6))
sns.scatterplot(x='number_of_reviews', y='price', data=df)
plt.title('Price vs. Number of Reviews')
plt.xlabel('Number of Reviews')
plt.ylabel('Price')
plt.show()

In [ ]:
room_type_avg_price = df.groupby('room_type')['price'].mean().reset_index()

plt.figure(figsize=(10, 6))
sns.barplot(x='room_type', y='price', data=room_type_avg_price)
plt.title('Average Price by Room Type')
plt.xlabel('Room Type')
plt.ylabel('Average Price')
plt.show()

# Questions & Answers

What is the average price of Airbnb listings in different neighborhoods?

In [ ]:
neighborhood_price = df.groupby('neighbourhood_group')['price'].mean().reset_index()
neighborhood_price.sort_values(by='price', ascending=False, inplace=True)

plt.figure(figsize=(10, 6))
sns.barplot(x='price', y='neighbourhood_group', data=neighborhood_price)
plt.title('Average Price by Neighborhood Group')
plt.xlabel('Average Price')
plt.ylabel('Neighborhood Group')
plt.show()


Which room type is the most common?

In [ ]:
room_type_count = df['room_type'].value_counts().reset_index()
room_type_count.columns = ['room_type', 'count']

plt.figure(figsize=(10, 6))
sns.barplot(x='count', y='room_type', data=room_type_count)
plt.title('Count of Room Types')
plt.xlabel('Count')
plt.ylabel('Room Type')
plt.show()


What is the average number of reviews per month for each neighborhood?

In [ ]:
# Average reviews per month per neighborhood
neighborhood_reviews = df.groupby('neighbourhood_group')['reviews_per_month'].mean().reset_index()
neighborhood_reviews.sort_values(by='reviews_per_month', ascending=False, inplace=True)

plt.figure(figsize=(10, 6))
sns.barplot(x='reviews_per_month', y='neighbourhood_group', data=neighborhood_reviews)
plt.title('Average Reviews per Month by Neighborhood Group')
plt.xlabel('Average Reviews per Month')
plt.ylabel('Neighborhood Group')
plt.show()


How does the type of room affect the price?

In [ ]:
room_type_avg_price = df.groupby('room_type')['price'].mean().reset_index()

plt.figure(figsize=(10, 5))
sns.barplot(x='room_type', y='price', data=room_type_avg_price)
plt.title('Average Price by Room Type')
plt.xlabel('Room Type')
plt.ylabel('Average Price')
plt.show()

Which neighborhoods have the highest and lowest number of listings?

In [ ]:
neighborhood_count = df['neighbourhood'].value_counts().reset_index()
neighborhood_count.columns = ['neighbourhood', 'listing_count']

top_neighborhoods = neighborhood_count.head(10)

plt.figure(figsize=(14, 7))
sns.barplot(x='listing_count', y='neighbourhood', data=top_neighborhoods)
plt.title('Top 10 Neighborhoods with Most Listings')
plt.xlabel('Number of Listings')
plt.ylabel('Neighborhood')
plt.show()

# Summary of Inferences

### Basic Statistics:

The dataset comprises various columns including price, number_of_reviews, reviews_per_month, availability_365, etc.
Basic statistics reveal that the average price of an Airbnb listing in NYC is around $152.72  with a standard deviation of $240.15. The minimum price is $0 and the maximum is $10,000, indicating a wide range of listing prices.

### Distribution of Price:

The distribution of prices is right-skewed, with the majority of listings priced under $500 per night A few high-end listings skew the distribution, resulting in a long tail on the right side

### Price vs Number of Reviews:

The scatter plot shows no strong correlation between the number of reviews and the price of a listing. However, it appears that more expensive listings tend to have fewer reviews, which might indicate that higher-priced listings are less frequently booked.

### Average Price by Neighborhood Group:

Manhattan has the highest average listing price among all neighborhood groups, followed by Brooklyn. Queens, Staten Island, and the Bronx have significantly lower average prices.
This suggests that listings in Manhattan and Brooklyn are generally more expensive, possibly due to higher demand and more central locations.

### Room Type Distribution:

Entire home/apartment is the most common room type, followed by private room. Shared room and hotel room types are relatively rare.
This distribution indicates a preference among Airbnb hosts to rent out entire properties rather than just individual rooms.

### Average Reviews per Month by Neighborhood Group:

Listings in the Bronx and Staten Island receive more reviews per month on average compared to other neighborhoods. Manhattan and Brooklyn, despite having higher listing prices, do not receive as many reviews per month.
This could imply that lower-priced listings in less expensive neighborhoods are more frequently booked, leading to more reviews.
### Availability Distribution:

The majority of listings are available for most of the year, with a significant number of listings having an availability of 365 days.
This suggests that many hosts keep their listings available year-round, possibly aiming to maximize their occupancy and rental income.

# Conclusion

The exploratory data analysis of the New York City Airbnb Open Data provides several insights into the characteristics and trends of Airbnb listings in NYC:

### Price Variability:

There is significant variability in listing prices, with a few high-end listings driving up the average. Most listings, however, are priced more affordably, catering to a wider audience.

### Neighborhood Trends:

Manhattan and Brooklyn are the most expensive neighborhoods for Airbnb listings, reflecting their popularity and central locations. In contrast, the Bronx, Queens, and Staten Island offer more affordable options, which may attract budget-conscious travelers.

### Room Type Preferences:

Entire home/apartment listings dominate the market, suggesting that both hosts and guests prefer more private accommodations. Private rooms also have a substantial presence, catering to solo travelers or those on a budget.

### Booking Frequency:

Listings in lower-priced neighborhoods like the Bronx and Staten Island tend to receive more frequent reviews, indicating higher booking rates. This could be due to their affordability and appeal to a broader range of travelers.

### Year-Round Availability:

Many listings are available year-round, indicating that hosts are keen to maximize their rental opportunities. This could also suggest a competitive market where hosts aim to attract guests consistently.


# Future Work

### Deeper Analysis of Guest Reviews:

Analyze the text of guest reviews to identify common themes, sentiments, and areas for improvement for hosts.

### Impact of Location on Booking Rates:

Investigate the impact of specific locations within neighborhoods on booking rates and prices. Proximity to landmarks, public transportation, and amenities could be influential factors.

### Seasonal Trends:

Explore seasonal trends in booking rates, prices, and availability to understand how different times of the year affect Airbnb listings in NYC.

### Comparison with Other Cities:

Compare the NYC Airbnb market with other major cities to identify unique trends and commonalities. This could provide valuable insights for hosts and travelers.

# Useful Resources

1. Pandas Documentation
2. Matplotlib Documentation
3. Seaborn Documentation
4. NumPy Documentation